diff --git a/backend/pnm.c b/backend/pnm.c
index 88b6e4a..c3acde4 100644
--- a/backend/pnm.c
+++ b/backend/pnm.c
@@ -219,7 +219,7 @@ static SANE_Option_Descriptor sod[] = {
    SANE_TYPE_FIXED,
    SANE_UNIT_PERCENT,
    sizeof (SANE_Word),
-   SANE_CAP_SOFT_SELECT | SANE_CAP_SOFT_DETECT,
+   SANE_CAP_SOFT_SELECT | SANE_CAP_SOFT_DETECT | SANE_CAP_COLOUR,
    SANE_CONSTRAINT_RANGE,
    {(SANE_String_Const *) & percentage_range}	/* this is ANSI conformant! */
    }
@@ -231,7 +231,7 @@ static SANE_Option_Descriptor sod[] = {
    SANE_TYPE_FIXED,
    SANE_UNIT_PERCENT,
    sizeof (SANE_Word),
-   SANE_CAP_SOFT_SELECT | SANE_CAP_SOFT_DETECT,
+   SANE_CAP_SOFT_SELECT | SANE_CAP_SOFT_DETECT | SANE_CAP_COLOUR,
    SANE_CONSTRAINT_RANGE,
    {(SANE_String_Const *) & percentage_range}	/* this is ANSI conformant! */
    }
@@ -243,7 +243,7 @@ static SANE_Option_Descriptor sod[] = {
    SANE_TYPE_BOOL,
    SANE_UNIT_NONE,
    sizeof (SANE_Word),
-   SANE_CAP_SOFT_SELECT | SANE_CAP_SOFT_DETECT,
+   SANE_CAP_SOFT_SELECT | SANE_CAP_SOFT_DETECT | SANE_CAP_COLOUR,
    SANE_CONSTRAINT_NONE,
    {NULL}
    }
@@ -321,7 +321,7 @@ static SANE_Option_Descriptor sod[] = {
    SANE_TYPE_BOOL,
    SANE_UNIT_NONE,
    sizeof (SANE_Word),
-   SANE_CAP_SOFT_SELECT | SANE_CAP_SOFT_DETECT,
+   SANE_CAP_SOFT_SELECT | SANE_CAP_SOFT_DETECT | SANE_CAP_COLOUR,
    SANE_CONSTRAINT_NONE,
    {NULL}
    }
@@ -333,7 +333,7 @@ static SANE_Option_Descriptor sod[] = {
    SANE_TYPE_INT,
    SANE_UNIT_NONE,
    sizeof (SANE_Word) * 256,
-   SANE_CAP_SOFT_SELECT | SANE_CAP_SOFT_DETECT | SANE_CAP_INACTIVE,
+   SANE_CAP_SOFT_SELECT | SANE_CAP_SOFT_DETECT | SANE_CAP_INACTIVE | SANE_CAP_COLOUR,
    SANE_CONSTRAINT_RANGE,
    {(SANE_String_Const *) & gamma_range}
    }
@@ -345,7 +345,7 @@ static SANE_Option_Descriptor sod[] = {
    SANE_TYPE_INT,
    SANE_UNIT_NONE,
    sizeof (SANE_Word) * 256,
-   SANE_CAP_SOFT_SELECT | SANE_CAP_SOFT_DETECT | SANE_CAP_INACTIVE,
+   SANE_CAP_SOFT_SELECT | SANE_CAP_SOFT_DETECT | SANE_CAP_INACTIVE | SANE_CAP_COLOUR,
    SANE_CONSTRAINT_RANGE,
    {(SANE_String_Const *) & gamma_range}
    }
@@ -357,7 +357,7 @@ static SANE_Option_Descriptor sod[] = {
    SANE_TYPE_INT,
    SANE_UNIT_NONE,
    sizeof (SANE_Word) * 256,
-   SANE_CAP_SOFT_SELECT | SANE_CAP_SOFT_DETECT | SANE_CAP_INACTIVE,
+   SANE_CAP_SOFT_SELECT | SANE_CAP_SOFT_DETECT | SANE_CAP_INACTIVE | SANE_CAP_COLOUR,
    SANE_CONSTRAINT_RANGE,
    {(SANE_String_Const *) & gamma_range}
    }
@@ -369,7 +369,7 @@ static SANE_Option_Descriptor sod[] = {
    SANE_TYPE_INT,
    SANE_UNIT_NONE,
    sizeof (SANE_Word) * 256,
-   SANE_CAP_SOFT_SELECT | SANE_CAP_SOFT_DETECT | SANE_CAP_INACTIVE,
+   SANE_CAP_SOFT_SELECT | SANE_CAP_SOFT_DETECT | SANE_CAP_INACTIVE | SANE_CAP_COLOUR,
    SANE_CONSTRAINT_RANGE,
    {(SANE_String_Const *) & gamma_range}
    }
diff --git a/include/sane/sane.h b/include/sane/sane.h
index 5320b4a..3fe82fc 100644
--- a/include/sane/sane.h
+++ b/include/sane/sane.h
@@ -115,6 +115,7 @@ SANE_Device;
 #define SANE_CAP_AUTOMATIC		(1 << 4)
 #define SANE_CAP_INACTIVE		(1 << 5)
 #define SANE_CAP_ADVANCED		(1 << 6)
+#define SANE_CAP_COLOUR		(1 << 7)
 
 #define SANE_OPTION_IS_ACTIVE(cap)	(((cap) & SANE_CAP_INACTIVE) == 0)
 #define SANE_OPTION_IS_SETTABLE(cap)	(((cap) & SANE_CAP_SOFT_SELECT) != 0)
